src/Controller/FilesController.php

```diff
@@ -78,16 +81,25 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        // Get the current user
+        $user = $this->security->getUser();
+
+        // Check if this is the home directory
+        $isHomeDir = $user instanceof FrontendUser && $user->homeDir === $filesModel->uuid;
```

Comment:
The `$isHomeDir` check needs to be done within the loop, otherwise it won't work for files in subdirectories of the home directory.